### PR TITLE
Prevent spawned Hardhat console from triggering a compile

### PR DESCRIPTION
**Pull request by kevinbluer**
## PR description

This PR prevents the [spawned Hardhat console](https://github.com/trufflesuite/truffle/blob/develop/packages/from-hardhat/src/ask-hardhat.ts#L76-L79) in `from-hardhat` from triggering a compile (via `--no-compile`). This is being added to mitigate a circular reference in an upcoming Hardhat plugin that provides support for Truffle Dashboard decoding/debugging.

## Testing instructions

One testing approach...

- Create a local Hardhat project (via `git clone https://github.com/amanusk/hardhat-template` or equivalent)
- `yarn link` to the `from-hardhat` package from the above
- Run the following from the `node` REPL:
  - `const FromHardhat = require("@truffle/from-hardhat");`
  - `await FromHardhat.prepareCompilations()`

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if documentation updates are required.

## Breaking changes and new features

- [ ] I have added any needed `breaking-change` and `new-feature` labels for the appropriate packages.

### CODE REVIEW

File: packages/from-hardhat/src/ask-hardhat.ts [hunk 1]
```diff
@@ -73,7 +73,7 @@ export const askHardhatConsole = async (
     const prefix = `${sos}truffle-start${st}`;
     const suffix = `${sos}truffle-end${st}`;
 
-    const hardhat = spawn(`npx`, ["hardhat", "console"], {
+    const hardhat = spawn(`npx`, ["hardhat", "console", "--no-compile"], {
```
Comment (cds-amal) [inline packages/from-hardhat/src/ask-hardhat.ts:76]: I think a comment explaining `--no-compile` is warranted here.

**Review by cds-amal — APPROVED** [review]
Looks good, @kevinbluer, though it would be prudent to comment why `--no-compile` is necessary for future maintainers.

Edit: tested with new hardhat project

File: packages/from-hardhat/src/ask-hardhat.ts (hunk 1 above)
Comment (kevinbluer) [reply packages/from-hardhat/src/ask-hardhat.ts:76]: 👍

**Comment by kevinbluer** [comment]
Comment added @cds-amal 👍 All good from your side @gnidan as I noticed the addition of the [breaking-change: @truffle/from-hardhat](https://github.com/trufflesuite/truffle/labels/breaking-change%3A%20%40truffle%2Ffrom-hardhat) label?

**Comment by cds-amal** [comment]
Hey @kevinbluer, the breaking change label is to communicate a breaking SemVer change (minor update) for the release process.

**Comment by kevinbluer** [comment]
Ah awesome 👍

**Review by gnidan — APPROVED** [review]
Yep all good now from my side! The label is just for housekeeping (for reasons stated)

**Pull request merged**